### DISCUSSION TITLE
Strong name signing WatsonWebserver

### DISCUSSION THREAD

**Discussion by bcalco** [dotnet/WatsonWebserver#85] [Q&A]
I want to use Watson in and environment that requires strongly named assemblies but it seems that the NuGet packaged version does not have a public key.  Is my only recourse to compile and sign it myself?

**Comment by jchristn** [comment]
Hi @bcalco at this point I don't have a financial incentive to do it (as far as I can tell, code signing certs aren't free), so it would be best for you to sign.  Cheers, Joel

**Comment by bcalco** [comment]
I was referring to simply strong-naming with an SNK file. That's pretty easy and costs nothing.

I am going to code-sign my binaries and the installer with an actual EV cert. I was hoping to avoid extra work of downloading, compiling and using Watson directly, when everything otherwise works just using Nuget to keep it current.

**Reply by jchristn** [reply]
Got it. Thought you were referring to signing. I will look into it ASAP. Thanks for the info!

**Reply by bcalco** [reply]
@jchristn Did you get a chance to look into adding an SNK file to the project and doing a release with it?

**Reply by jchristn** [reply]
Yes - I'm doing it now - question for you, as a best practice, I assume I should ```.gitignore``` the ```.snk``` file?

**Reply by bcalco** [reply]
I've seen it done both ways, but as a best practice probably best to gitignore it. Anybody recompiling would have to supplie their own, which has the benefit then of being able to cmpare the public key so you can say for sure whether it was a binary you compiled or not. The SNK files associated with strong-naming dont really add much by way of security, per se, beyond that. Kinda like self-signed certs. When it really matters, you sign the binaries with a secure "real" cert, which you would never put in git anyway. :)

**Reply by bcalco** [reply]
Some open source projects with more than one repo even have a repo just for the "shared" SNK file used across them. It depends on your motivation and what you're trying to achieve with it. Being able to run in environments that require a strong name is motivation enough for the ceremony, I guess.

**Reply by jchristn** [reply]
Applying strong naming to the subordinate packages, then will do Watson, give me a few mins

**Reply by jchristn** [reply]
Hi @bcalco I just uploaded v4.2.2.6, can you let me know if this does the trick?  https://www.nuget.org/packages/Watson/4.2.2.6

**Reply by bcalco** [reply]
Will check it out and let you know. It may or may get rid of problem I was seeing trying to load it (may have a different root cause), but it should minimally get rid of the missing strong name warning. :)

**Reply by bcalco** [reply]
OK it gets rid of the warning, and it does seem I'm able to load the assembly in my app and call into it. Thanks!

**Reply by jchristn** [reply]
Thanks for the pointers and the nudge.  Not sure how I'm going to navigate doing this on the rest of my packages lol :)
